src/components/timeline/TrackContextMenu.js

```diff
@@ -317,6 +319,71 @@ class TimelineTrackContextMenuImpl extends PureComponent<
     }
   };
 
+  _getRightClickedTrackType = (): string => {
+    const { rightClickedTrack, localTracksByPid, globalTracks } = this.props;
+
+    if (rightClickedTrack === null) {
+      throw new Error(
+        'Attempted to isolate the process with no right clicked track.'
+      );
+    }
+
+    let track;
+    if (rightClickedTrack.type === 'local') {
+      const localTracks = ensureExists(
+        localTracksByPid.get(rightClickedTrack.pid)
+      );
+      track = localTracks[rightClickedTrack.trackIndex];
+    } else {
+      track = globalTracks[rightClickedTrack.trackIndex];
+    }
+
+    return track.type;
+  };
+
+  _hideTracksByType = (_): void => {
+    const {
+      rightClickedTrack,
+      globalTracks,
+      globalTrackNames,
+      localTracksByPid,
+      localTrackNamesByPid,
+      threads,
+      hideProvidedTracks,
+    } = this.props;
+    const type = this._getRightClickedTrackType();
+
+    if (rightClickedTrack === null) {
+      throw new Error(
+        'Attempted to isolate the process with no right clicked track.'
```

Comment:
nit: I see you copied this from another function, but can you please update the error message?